pkg/ccl/changefeedccl/helpers_test.go

```diff
@@ -1629,3 +1631,68 @@ func checkSchema(actual []cdctest.TestFeedMessage) error {
 	}
 	return nil
 }
+
+// runWithAndWithoutRegression141453 runs the test both with and without testing
+// knobs that simulate the scenario where a change aggregator encounters a schema
+// change restart but draining the buffer fails so the resolved spans message
+// signaling the restart doesn't get sent to the change frontier.
+func runWithAndWithoutRegression141453(
+	t *testing.T, testFn cdcTestFn, runTestFn func(t *testing.T, testFn cdcTestFn),
+) {
+	testutils.RunTrueAndFalse(t, "regression 141453",
+		func(t *testing.T, regression141453 bool) {
+			testFn := func(t *testing.T, s TestServer, f cdctest.TestFeedFactory) {
+				if !regression141453 {
+					testFn(t, s, f)
+					return
+				}
+
+				knobs := s.TestingKnobs.
+					DistSQL.(*execinfra.TestingKnobs).
+					Changefeed.(*TestingKnobs)
+
+				// We only want to make drain fail once, otherwise the test will never
```

Comment:
nit: can you expand this comment to describe exactly what we're doing here? iirc it waits till an event is Added, then makes subsequent Pops block until Drain has finished